subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/verification/writer/WriteDependencyVerificationFile.java

```diff
@@ -437,6 +437,13 @@ public void onArtifact(ArtifactKind kind, ModuleComponentArtifactIdentifier id,
         }
     }
 
+    @Override
+    public boolean wasAlreadyProcessed(ModuleComponentArtifactIdentifier artifact, String repositoryId) {
+        // Since writing to a file is done rarely and this is called only to avoid resolving
+        // artifacts again when already cached, there is not much penalty to not do any check here
+        return false;
```

Comment:
I wonder if that statement is true indeed.
When writing a dependency verification file, we are most often resolving _all_ configurations in a project. If we start re-resolving artifacts every time, this could have quite an effect on resolution time.
Did you validate your assumption on the `gradle/gradle` build for example?

Reply:
I will check with `gradle/gradle`.

Reply:
It doesn't look much difference for 
`gradle --write-verification-metadata pgp,sha256 help --no-configuration-cache `:
<img width="1988" alt="Screenshot 2022-01-18 at 19 38 40" src="https://user-images.githubusercontent.com/3939893/149998536-7401f518-24c7-4fc6-8b7b-3d3498379a18.png">

Once artifacts are in cache this is quick.

The bigger issue for me was that at the first start I got `https://keys.fedoraproject.org` keyserver that seem to be dead and my build took 20 minutes+ due to retries before I killed it.

Reply:
Opened https://github.com/gradle/gradle/issues/19605 issue to remove `keys.fedoraproject.org`.

Reply:
Thanks for looking and taking care of that other issue.

I would wait to merge this, as we may want to target release if we end up doing 7.4 RC2